File: src/main/java/kz/nearbygems/chat/service/impl/ChatServiceImpl.java
package kz.nearbygems.chat.service.impl;

import io.netty.channel.ChannelHandlerContext;
import kz.nearbygems.chat.model.Command;
import kz.nearbygems.chat.provider.CommandProviderSelector;
import kz.nearbygems.chat.service.ChatService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class ChatServiceImpl implements ChatService {

    private final CommandProviderSelector commands;

    @Override
    public void handle(ChannelHandlerContext ctx, String msg) {

        if (msg.isEmpty()) {
            return;
        }

        final var command = Command.Companion.from(ctx, msg);

        commands.select(command)
                .execute(command);
    }

}

File: src/main/java/kz/nearbygems/chat/service/impl/ChatServiceImpl.kt
package kz.nearbygems.chat.service.impl

import kz.nearbygems.chat.repository.ChatRepository
import kz.nearbygems.chat.service.ChatService
import org.springframework.stereotype.Service

@Service
class ChatServiceImpl(private val repository: ChatRepository) : ChatService {

    override fun getChatNameByUsername(username: String): String? =
        repository.getChatNameByUsername(username)

    override fun getUsernamesByChatName(chatName: String): List<String> =
        repository.getUsernamesByChatName(chatName)

    override fun saveChatName(username: String, chatName: String) {
        repository.saveChatName(username, chatName)
    }

    override fun deleteChatName(username: String) {
        repository.deleteChatName(username)
    }

}